ACG/AutoRouterShield.py

```diff
@@ -0,0 +1,714 @@
+from .AyarLayoutGenerator import AyarLayoutGenerator
+from .Rectangle import Rectangle
+from .XY import XY
+from typing import Tuple, Union, Optional, List, Dict
+from .AutoRouter import EZRouter
+
+
+class EZRouterShield(EZRouter):
+    """
+    The EZRouterShield class inherits from the EZRouter class and allows you to create ground-shielded routes.
+    """
+
+    def __init__(self,
+                 gen_cls: AyarLayoutGenerator,
+                 start_rect: Optional[Rectangle] = None,
+                 start_direction: Optional[str] = None,
+                 config: Optional[dict] = None
+                 ):
+        EZRouter.__init__(self, gen_cls, start_rect=start_rect, start_direction=start_direction, config=config)
+
+        self.route_points = []
+        self.route_point_dict = {}
+
+        # to determine offset of shield_1 from center
+        self.shield_dict = {
+            '+x': {
+                '+x': (0, 1),
+                '-x': (0, 1),
+                '+y': (-1, 1),
+                '-y': (1, 1)
+            },
+            '-x': {
+                '+x': (0, -1),
+                '-x': (0, -1),
+                '+y': (-1, -1),
+                '-y': (1, -1)
+            },
+            '+y': {
+                '+x': (-1, 1),
+                '-x': (-1, -1),
+                '+y': (-1, 0),
+                '-y': (-1, 0)
+            },
+            '-y': {
+                '+x': (1, 1),
+                '-x': (1, -1),
+                '+y': (1, 0),
+                '-y': (1, 0)
+            }
+        }
+
+    def add_route_points(self,
+                         points: List[Tuple],
+                         metal_layer: str,
```

Comment:
We should try and change variable names like `metal_layer`, `start_met`, etc. to just `layer` just to be generic.